lib/components/admin/mailables-window.js

```diff
@@ -0,0 +1,194 @@
+import React, { Component } from 'react'
+import { Badge } from 'react-bootstrap'
+import { connect } from 'react-redux'
+
+import * as callTakerActions from '../../actions/call-taker'
+import DraggableWindow from './draggable-window'
+import Icon from '../narrative/icon'
+import {MailablesList, WindowHeader} from './styled'
+import {createLetter, MAILABLE_FIELDS} from '../../util/mailables'
+
+/**
+ * A window enabled through the Mailables module that allows Call Taker users
+ * to generate a PDF with an invoice of items to be mailed to transit customers.
+ */
+class MailablesWindow extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      mailables: []
+    }
+  }
+
+  _addMailable = (mailable) => {
+    if (!this.state.mailables.find(m => m.name === mailable.name)) {
+      const mailables = [...this.state.mailables]
+      mailables.push({...mailable, quantity: 1})
+      this.setState({mailables})
+    }
+  }
+
+  _removeMailable = (mailable) => {
+    const mailables = [...this.state.mailables]
+    const removeIndex = mailables.findIndex(m => m.name === mailable.name)
+    mailables.splice(removeIndex, 1)
+    this.setState({mailables})
+  }
+
+  _updateMailableField = (index, fieldName, value) => {
+    const mailables = [...this.state.mailables]
+    mailables[index] = {...mailables[index], [fieldName]: value}
+    this.setState({mailables})
+  }
+
+  _onClickCreateLetter = () => createLetter(this.state, this.props.mailablesConfig)
+
+  _updateField = (evt) => {
+    this.setState({[evt.target.id]: evt.target.value})
+  }
+
+  render () {
+    const {mailablesConfig, callTaker, toggleMailables} = this.props
+    const {mailables: selectedMailables} = this.state
+    const {items} = mailablesConfig
+    if (!callTaker.mailables.visible) return null
+    const selectableMailables = items.filter(m =>
+      !selectedMailables.find(mailable => mailable.name === m.name)
+    )
+    return (
+      <DraggableWindow
+        header={
+          <>
+            <WindowHeader>
+              <Icon type='graduation-cap' /> Mailables{' '}
+              <span className='pull-right'>
+                <button
+                  className='clear-button-formatting'
+                  onClick={this._onClickCreateLetter}
+                  style={{marginRight: '5px', verticalAlign: 'bottom'}}
+                >
+                  Create Letter
+                </button>
+              </span>
+            </WindowHeader>
+          </>
+        }
+        onClickClose={toggleMailables}
+        scroll={false}
+        style={{height: '340px', width: '600px'}}
+      >
+        <div>
+          {MAILABLE_FIELDS.map(f => (
```

Comment:
Would this be better called LETTER_FIELDS to distinguish it from being a mailable item?